app/services/tag_adjustment_creation_service.rb

```diff
@@ -20,14 +20,30 @@ def update_tags_and_notify
   private
 
   def update_article
-    article.update!(tag_list: article.tag_list.remove(@tag_adjustment.tag_name)) if @tag_adjustment.adjustment_type == "removal"
+    if @tag_adjustment.adjustment_type == "removal"
+      removed_tags = article.tag_list.select { |tag| tag.casecmp(@tag_adjustment.tag_name).zero? }
+      return if removed_tags.empty?
+
+      article.update!(tag_list: article.tag_list.remove(removed_tags))
+    end
+
     article.update!(tag_list: article.tag_list.add(@tag_adjustment.tag_name)) if @tag_adjustment.adjustment_type == "addition"
   end
 
   def creation_args
     args = @tag_adjustment_params
+
+    tag =
+      case args[:adjustment_type]
+      when "removal"
+        article.tags.detect { |article_tag| article_tag.name.casecmp(args[:tag_name]).zero? }
```

Comment:
This is done to do an efficient case insensitive search - we just grab the article's tag records (max 4) and loop through them, ignoring case.

This was done instead of adding an index to the `tags` table to do a `lower(name)` query. This is all going to change once we _actually_ fix tags.